### PR TITLE
network: modernize routing policy rule and add 0 validation where needed

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/shared/conf-parser.c [hunk 1]
```diff
@@ -1143,7 +1143,13 @@ int config_parse_mtu(
                            "Failed to parse MTU value '%s', ignoring: %m", rvalue);
                 return 0;
         }
+        if (k == 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Invalid MTU value '%s', ignoring: %m", rvalue);
+                return 0;
+        }
 
+        *mtu = k;
```
Comment (yuwata) [inline src/shared/conf-parser.c]: This is not necessary, as `parse_mtu()` already checked its validity. Please drop the second commit.

**Review by yuwata — CHANGES_REQUESTED** [review]
The first and third commit is OK. Please drop the second commit.

**Comment by ssahani** [comment]
Updated thanks for the review.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**